cedar-lean/Cedar/Spec/Ext/Decimal.lean

```diff
@@ -58,6 +58,28 @@ def parse (str : String) : Option Decimal :=
     else .none
   | _ => .none
 
+def unParse (d : Decimal) : String :=
+  let neg   := if d < (0 : Int) then "-" else ""
+  let d     := d.natAbs
+  let left  := d / (Nat.pow 10 DECIMAL_DIGITS)
+  let right := d % (Nat.pow 10 DECIMAL_DIGITS)
+  let right :=
+    -- this is not generalized for arbitrary DECIMAL_DIGITS
+    if right < 10 then s!".000{right}"
+    else if right < 100 then s!".00{right}"
+    else if right < 1000 then s!".0{right}"
+    else s!".{right}"
+  s!"{neg}{left}{right}"
+
+theorem test1 : unParse ((parse "3.14").get!) = "3.1400" := by decide
```

Comment:
I suggest moving these to `UnitTest.Decimal`: https://github.com/cedar-policy/cedar-spec/blob/main/cedar-lean/UnitTest/Decimal.lean

Reply:
Moved (and the ip tests)